crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -208,12 +223,16 @@ impl ConfigManagerActor {
             .plugin_config
             .get_file_entry_from_type(&request.config_type)?;
 
-        let upload_request =
-            UploadRequest::new(&request.tedge_url, Utf8Path::new(&file_entry.path));
+        let Some(tedge_url) = &request.tedge_url else {
+            debug!("tedge_url not present in config snapshot payload, ignoring");
```

Comment:
I'm thinking if ignoring this is the right behaviour or not.

1. tedge_config_manager adds `tedgeUrl` to the payload if not provided when status changes from `init` to `executing`.
2. It means, if the status is `executing` but `tedgeUrl` is not ptovided, the command is created not by tedge_config_manager.

So ignoring is correct because the command is not meant for tedge_config_manager?
Or should be an error and mark the command failed because tedgeUrl is not provided in executing state?
Or even tedge_config_manager tries to add `tedgeUrl` if `tedgeUrl` is not given here?

When we decide, it's worth adding a unit test.

Reply:
Ok, I talked with @albinsuresh, and we concluded here.

If `tedgeUrl` is not provided in this state, tedge-agent adds `tedgeUrl` here.

I definitely recommend to create a unit test to prove this story.

Reply:
Resolved by fd79b06104b4abc19853b565609d3abc103267c8